### PR TITLE
Add NATS-manager module build job

### PR DESCRIPTION
**Pull request by mfaizanse**
**Description**

Changes proposed in this pull request:

- Add NATS-manager module build job
  - `pull-nats-module-build` - [sample run](https://status.build.kyma-project.io/view/gs/kyma-prow-logs/pr-logs/pull/kyma-project_nats-manager/82/mfaizanse_test_of_prowjob_pull-nats-module-build/1680827668452347904)
  - `post-nats-module-build` - [sample run](https://status.build.kyma-project.io/view/gs/kyma-prow-logs/logs/mfaizanse_test_of_prowjob_post-nats-module-build/1680827663213662208)
- Changed `run_if_changed` for docker image build job because it needs to run when `resources` are changed as well.

**Related issue(s)**

- https://github.com/kyma-project/nats-manager/issues/75

### CODE REVIEW

File: prow/jobs/nats-manager/nats-manager-generic.yaml [hunk 1]
```diff
@@ -13,7 +13,7 @@ presubmits: # runs on PRs
         prow.k8s.io/pubsub.runID: "pull-nats-manager-build"
         prow.k8s.io/pubsub.topic: "prowjobs"
         preset-sa-kyma-push-images: "true"
-      run_if_changed: '^Dockerfile|\.go$|^go.(sum|mod)'
+      run_if_changed: '^.env$|^hack/ci/*/.*$|^pkg/*/.*$|^internal/*/.*$|^cmd/*/.*$|^resources/*/.*$|^api/*/.*$|^Dockerfile$|^(go.mod|go.sum)$|^*/(.*.go|Makefile|.*.sh)|^PROJECT$|^config/*/.*$|^docs/*/.*$'
```
Comment (marcobebway) [inline prow/jobs/nats-manager/nats-manager-generic.yaml:16]: Why not run it if anything changed?

Also shouldn't `/*/` change to `/.*/`? or did I miss something?

File: prow/jobs/nats-manager/nats-manager-generic.yaml [hunk 2]
```diff
@@ -82,6 +82,50 @@ presubmits: # runs on PRs
...
+      annotations:
+        description: "nats module build pre-main job"
+        owner: "skydivingtunas"
+        pipeline.trigger: "pr-submit"
+      labels:
+        prow.k8s.io/pubsub.project: "sap-kyma-prow"
+        prow.k8s.io/pubsub.runID: "pull-nats-module-build"
+        prow.k8s.io/pubsub.topic: "prowjobs"
+        preset-sa-kyma-push-images: "true"
+      run_if_changed: '^.env$|^hack/ci/*/.*$|^pkg/*/.*$|^internal/*/.*$|^cmd/*/.*$|^resources/*/.*$|^api/*/.*$|^Dockerfile$|^(go.mod|go.sum)$|^*/(.*.go|Makefile|.*.sh)|^PROJECT$|^config/*/.*$|^docs/*/.*$'
```
Comment (marcobebway) [inline prow/jobs/nats-manager/nats-manager-generic.yaml:95]: Same as https://github.com/kyma-project/test-infra/pull/8060/files#r1264989673.

File: prow/jobs/nats-manager/nats-manager-generic.yaml [hunk 3]
```diff
@@ -82,6 +82,50 @@ presubmits: # runs on PRs
               requests:
                 memory: 1Gi
                 cpu: 100m
+    - name: pull-nats-module-build
+      annotations:
+        description: "nats module build pre-main job"
```
Comment (marcobebway) [inline prow/jobs/nats-manager/nats-manager-generic.yaml:87]:
```suggestion
        description: "NATS module build pre-main job"
```

File: prow/jobs/nats-manager/nats-manager-generic.yaml [hunk 4]
```diff
@@ -136,4 +180,48 @@ postsubmits: # runs on main
           - name: signify-secret
             secret:
               secretName: signify-dev-secret
+    - name: post-nats-module-build
+      annotations:
+        description: "NATS module build job"
```
Comment (marcobebway) [inline prow/jobs/nats-manager/nats-manager-generic.yaml:185]:
```suggestion
        description: "NATS module build post-main job"
```

**Review by marcobebway — CHANGES_REQUESTED** [review]
The run for [post-nats-module-build](https://status.build.kyma-project.io/view/gs/kyma-prow-logs/logs/mfaizanse_test_of_prowjob_post-nats-module-build/1680827663213662208) failed. Is this expected?

File: prow/jobs/nats-manager/nats-manager-generic.yaml (hunk 3 above)
Comment (mfaizanse) [reply prow/jobs/nats-manager/nats-manager-generic.yaml]: updated

File: prow/jobs/nats-manager/nats-manager-generic.yaml (hunk 4 above)
Comment (mfaizanse) [reply prow/jobs/nats-manager/nats-manager-generic.yaml]: /updated

File: prow/jobs/nats-manager/nats-manager-generic.yaml [hunk 5]
```diff
@@ -60,7 +60,7 @@ presubmits: # runs on PRs
         prow.k8s.io/pubsub.project: "sap-kyma-prow"
         prow.k8s.io/pubsub.runID: "pull-nats-manager-unit-test"
         prow.k8s.io/pubsub.topic: "prowjobs"
-      run_if_changed: '^\S+\.go$|^go\.(sum|mod)$'
+      skip_if_only_changed: '^docs/|\.(md|adoc)$|^(README|LICENSE)$'
```
Comment (mfaizanse) [inline prow/jobs/nats-manager/nats-manager-generic.yaml:63]: https://docs.prow.k8s.io/docs/jobs/#triggering-jobs-based-on-changes

File: prow/jobs/nats-manager/nats-manager-generic.yaml (hunk 1 above)
Comment (mfaizanse) [reply prow/jobs/nats-manager/nats-manager-generic.yaml]: made post-main jobs to always run, and pre-main to skip only for docs.

**Review by marcobebway — APPROVED** [review]

**Pull request merged**